sdk/python/feast/feature_store.py

```diff
@@ -161,14 +185,29 @@ def list_feature_services(self) -> List[FeatureService]:
         return self._registry.list_feature_services(self.project)
 
     @log_exceptions_and_usage
-    def list_feature_views(self) -> List[FeatureView]:
+    def list_feature_views(self, allow_cache: bool = False) -> List[FeatureView]:
         """
         Retrieves the list of feature views from the registry.
 
+        Args:
+            allow_cache: Whether to allow returning entities from a cached registry.
+
         Returns:
             A list of feature views.
         """
-        return self._registry.list_feature_views(self.project)
+        return self._list_feature_views(allow_cache)
+
+    def _list_feature_views(
+        self, allow_cache: bool = False, hide_dummy_entity: bool = True
+    ) -> List[FeatureView]:
+        feature_views = []
+        for fv in self._registry.list_feature_views(
+            self.project, allow_cache=allow_cache
+        ):
+            if hide_dummy_entity and fv.entities[0] == DUMMY_ENTITY_NAME:
+                fv.entities = []
+            feature_views.append(fv)
```

Comment:
Is there the possibility that we return to `feature_views`, but this list gets re-applied using `FeatureStore.apply` and we lose the entity information?